packages/react-components/src/hooks/useTimeSeriesData/providerStore.ts

```diff
@@ -0,0 +1,25 @@
+import { ProviderWithViewport, TimeSeriesData } from '@iot-app-kit/core';
+
+type ProviderMap = { [key in string]: ProviderWithViewport<TimeSeriesData[]> };
+const providerStore = () => {
+  const providerMap: ProviderMap = {};
+
+  const set = (id: keyof ProviderMap, provider: ProviderMap[keyof ProviderMap]) => {
+    providerMap[id] = provider;
+    return provider;
+  };
+
+  const get = (id: keyof ProviderMap): ProviderMap[keyof ProviderMap] | undefined => providerMap[id];
+
+  const remove = (id: keyof ProviderMap) => {
+    delete providerMap[id];
+  };
+
+  return {
+    get,
+    set,
+    remove,
+  };
+};
+
+export const ProviderStore = providerStore();
```

Comment:
Why not make an object literal, instead of a function, which is a singleton? Are you taking advantage of the closure around providerMap in a way that's not achievable with a plain object?